spellfix.py

```diff
@@ -99,41 +117,69 @@ def show_unknown(self):
         return self.unknown.word_frequency.dictionary
 
+    def show_skipped(self):
+        """
+        """
+        return self.skipped
+
+    def skip(self):
+        """
+        Add word to skip list and remove from unknown words.
+        """
+        print("Skipping %s"%self.word)
+        self.skipped[self.word] = self.unknown.word_frequency.dictionary[self.word]
+        self.unknown.word_frequency.remove(self.word)
+
+    def dump_skipped(self):
+        for word in self.skipped:
+            for _ in range(self.skipped[word]):
+                self.unknown.word_frequency.add(word)
+        self.skipped = {}
+        print("Dumped skipped back into unknown")
+
     def correct(self):
         """
         """
         quit = False
         uwfq = self.unknown.word_frequency
         kwfq = self.known.word_frequency
         unknown_words = list(uwfq.dictionary.keys())
+        suggest = self.suggest
+        
         if len(unknown_words) == 0:
-            quit = True
-            print("YOU ARE DONE!")
-            return quit # FINISHED!
+            if len(self.skipped) == 0:
+                quit = True
+                print("YOU ARE DONE!")
+                self.save()
+                return quit # FINISHED!
+            else:
+                print("Almost done! Moving skipped into unknown.")
+                self.dump_skipped()
+                return quit # CYCLE BACK THROUGH AGAIN
         else:
             r = len(unknown_words)
             word = ''
-            j = 0 # how many times can we try passing through short words before giving up?
-            while (len(word) <= 4) and (j<10):
-                word = unknown_words[random.randint(0,r-1)]
-                j += 1
-
+            self.word = word
+            if self.random_choice:
+                j = 0 # how many times can we try passing through short words before giving up?
+                while (len(word) <= 4) and (j<10):
+                    word = unknown_words[random.randint(0,r-1)]
+                    j += 1
+            else:
+                word = unknown_words[0]
+                self.word = word # store for reference
+            if len(word)>10:
+                self.save()
         #self.unknown.word_frequency.remove(word)
         # uwfq.remove(word)
         print("\t===> %s"%word)
         known_candidates = list(self.known.candidates(word))
         unknown_candidates = list(self.unknown.candidates(word))
-        fname = self.prefix + '-matches/' + format_str(word) + '.csv'
-        try:
-            with open(fname, 'r') as f:
-                loaded_options = f.read().splitlines()
-            # clean up duplicates from loaded list.
-            for w in known_candidates + unknown_candidates:
-               if w in loaded_options:
-                    loaded_options.remove(w) 
-            unknown_candidates += loaded_options
-        except FileNotFoundError: # sometimes the TFidF doesn't match words
-            loaded_options = []
```

Comment:
this is the chunk of code we can re-introduce if `suggest` is empty.

Reply:
and since we're removing duplicates anyway, it's not a problem if both get loaded. The logic can be "if csv files exist, use them to enrich suggestions" rather than "try to load CSV if suggestion dictionary is missing"